dlt/destinations/impl/filesystem/filesystem.py

```diff
@@ -123,16 +124,51 @@ def write(self) -> None:
         from dlt.common.libs.deltalake import (
             write_delta_table,
             _deltalake_storage_options,
+            try_get_deltatable,
         )
 
         file_paths = [job.file_path for job in self.table_jobs]
 
-        write_delta_table(
-            path=self.client.make_remote_uri(self.make_remote_path()),
-            data=pa.dataset.dataset(file_paths),
-            write_disposition=self.table["write_disposition"],
-            storage_options=_deltalake_storage_options(self.client.config),
-        )
+        if (
```

Comment:
This should probably be refactored to match how we do it for SQL merge jobs:
- use separate class for merge job
- process entire table chain in job

Reply:
I don't think it is 100% necessary to have a separate merge job, but processing all table chain jobs at once maybe yes. This is kind of there already in my loader changes, so I think you can have a look at the new structure once it is done and figure it out.